server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -451,6 +459,41 @@ export class AssetService {
     return this._assetRepository.getExistingAssets(authUser.id, checkExistingAssetsDto);
   }
 
+  async checkExistenceOfAssetsByChecksum(
+    authUser: AuthUserDto,
+    checkExistenceOfAssetsDto: CheckExistenceOfAssetsByChecksumDto,
+  ): Promise<CheckExistenceOfAssetsResponseDto> {
+    const queriedChecksums: Buffer[] = checkExistenceOfAssetsDto.assets.map((asset) =>
+      Buffer.from(asset.checksum, 'hex'),
+    );
+
+    const existingAssets: Promise<AssetEntity[]> = this._assetRepository.getAssetsByChecksums(
```

Comment:
I'm new to ts, but wouldn't that block for every array element? I thought this is faster since it just blocks once, allowing all elements to finish asynchronously

Reply:
No it wouldn't block for every element. The repository will make one database call and return all elements. You can see the return type for getAssetsByChecksums is `Promise<AssetEntity[]>`. You can just await the Promise and the type will become `AssetEntity[]`.

If the return type would be `Promise<AssetEntity>[]` then you have to await each individual element.